lib/stacks/api.ts

```diff
@@ -0,0 +1,437 @@
+    super(scope, id, props);
+
+    const stage = this.node.tryGetContext("stage") || "dev";
+
+    const service = "app-api";
+    this.shortStackName = `${service}-${stage}`;
+    cdk.Tags.of(this).add("SERVICE", service);
+
+    this.tables = props.tables;
+
+    const { vpc, privateSubnets, isDev, brokerString } = props;
+
+    const kafkaSecurityGroup = new ec2.SecurityGroup(
+      this,
+      "KafkaSecurityGroup",
+      {
+        vpc,
+        description:
+          "Security Group for streaming functions. Egress all is set by default.",
+        allowAllOutbound: true,
+      }
+    );
+
+    const logGroup = new LogGroup(this, "ApiAccessLogs", {
+      logGroupName: `/aws/api-gateway/${stage}-app-api`,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+    });
+
+    this.api = new apigateway.RestApi(this, "ApiGatewayRestApi", {
+      restApiName: `${stage}-app-api`,
+      deploy: true,
+      deployOptions: {
+        stageName: stage,
+        tracingEnabled: true,
+        loggingLevel: apigateway.MethodLoggingLevel.INFO,
+        dataTraceEnabled: true,
+        metricsEnabled: false,
+        throttlingBurstLimit: 5000,
+        throttlingRateLimit: 10000.0,
+        cachingEnabled: false,
+        cacheTtl: cdk.Duration.seconds(300),
+        cacheDataEncrypted: false,
+        accessLogDestination: new apigateway.LogGroupLogDestination(logGroup),
+        accessLogFormat: apigateway.AccessLogFormat.custom(
+          "requestId: $context.requestId, ip: $context.identity.sourceIp, " +
+            "caller: $context.identity.caller, user: $context.identity.user, " +
+            "requestTime: $context.requestTime, httpMethod: $context.httpMethod, " +
+            "resourcePath: $context.resourcePath, status: $context.status, " +
+            "protocol: $context.protocol, responseLength: $context.responseLength"
+        ),
+      },
+      defaultCorsPreflightOptions: {
+        allowOrigins: apigateway.Cors.ALL_ORIGINS,
+        allowMethods: apigateway.Cors.ALL_METHODS,
+      },
+    });
+
+    this.api.addGatewayResponse("Default4XXResponse", {
+      type: apigateway.ResponseType.DEFAULT_4XX,
+      responseHeaders: {
+        "Access-Control-Allow-Origin": "'*'",
+        "Access-Control-Allow-Headers": "'*'",
+      },
+    });
+
+    this.api.addGatewayResponse("Default5XXResponse", {
+      type: apigateway.ResponseType.DEFAULT_5XX,
+      responseHeaders: {
+        "Access-Control-Allow-Origin": "'*'",
+        "Access-Control-Allow-Headers": "'*'",
+      },
+    });
+
+    const commonProps = {
+      brokerString,
+      stackName: this.shortStackName,
+      tables: this.tables,
+      api: this.api,
+    };
+
+    new Lambda(this, "ForceKafkaSync", {
+      entry: "services/app-api/handlers/kafka/get/forceKafkaSync.js",
+      timeout: cdk.Duration.minutes(15),
+      memorySize: 3072,
+      ...commonProps,
+    });
+
+    const postKafkaData = new Lambda(this, "postKafkaData", {
```

Comment:
We should check this is still running successfully when we're editing data in the app.

The logs are extremely chatty when it can't connect or something goes wrong, so just vetting that postKafkaData is returning successes.